src/main/clm_varpar.F90

```diff
@@ -227,25 +231,33 @@ subroutine clm_varpar_init(actual_maxsoil_patches, actual_numcft)
        write(iulog, *)
     end if
 
+    ! We hardwire these parameters here because we use them
+    ! in InitAllocate (in SoilBiogeochemStateType) which is called earlier than
+    ! init_decompcascade_bgc where they might have otherwise been derived on the
+    ! fly. For reference, if they were determined in init_decompcascade_bgc:
+    ! ndecomp_pools would get the value of i_avl_som or i_cwd and
+    ! ndecomp_cascade_transitions would get the value of i_s3s1 or i_cwdl3
+    ! depending on how use_fates is set.
     if ( use_fates ) then
-       i_cwd = 0
        if (use_century_decomp) then
           ndecomp_pools = 6
           ndecomp_cascade_transitions = 8
-       else
+       else  ! TODO slevis: Looks like CN to me, so plan on removing?
```

Comment:
Yes, this is the CN specific else. So this would be removed eventually. Although we still need a trigger to choose between the different available decomposition options. Now it's CN and BGC, but will be BGC or MIMICS. So we do want to retain the ability to select more than one decomp option.